functests/ptp/ptp.go

```diff
@@ -108,4 +93,417 @@ var _ = Describe("ptp", func() {
 			}
 		})
 	})
+	var _ = Describe("PTP e2e tests", func() {
+		var ptpRunningPods []v1core.Pod
+		var _ = Context("PTP Interfaces discovery", func() {
+
+			BeforeEach(func() {
+				var err error
+				ptpRunningPods = []v1core.Pod{}
+				ptpPods, err := client.Client.Pods(ptpLinuxDaemonNamespace).List(metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(ptpPods.Items)).To(BeNumerically(">", 0), fmt.Sprint("linuxptp-daemon is not deployed on cluster"))
+				for _, pod := range ptpPods.Items {
+					if podRole(pod, ptpSlaveNodeLabel) || podRole(pod, ptpGrandmasterNodeLabel) {
+						waitUntilLogIsDetected(pod, 180, "PTP capable NICs")
+						ptpRunningPods = append(ptpRunningPods, pod)
+					}
+				}
+				Expect(len(ptpRunningPods)).To(BeNumerically(">", 0), fmt.Sprint("Fail to detect PTP slave/master pods on Cluster"))
+			})
+
+			// 25729
+			It("The interfaces support ptp can be discovered correctly", func() {
+
+				for _, pod := range ptpRunningPods {
+					ptpSupportedInt := getPtpMasterSlaveAttachedInterfaces(pod)
+					Expect(len(ptpSupportedInt)).To(BeNumerically(">", 0), fmt.Sprint("Fail to detect PTP Supported interfaces on slave/master pods"))
+					ptpDiscoveredInterfaces := ptpDiscoveredInterfaceList(nodePtpDeviceAPIPath + pod.Spec.NodeName)
+					Expect(len(ptpSupportedInt)).To(BeNumerically("==", len(ptpDiscoveredInterfaces)), fmt.Sprint("The interfaces discovered incorrectly"))
+					for _, intfc := range ptpSupportedInt {
+						Expect(ptpDiscoveredInterfaces).To(ContainElement(intfc))
+					}
+				}
+			})
+
+			// 25730
+			It("The virtual interfaces should be not discovered by ptp", func() {
+
+				for _, pod := range ptpRunningPods {
+					ptpNotSupportedInt := getNonPtpMasterSlaveAttachedInterfaces(pod)
+					ptpDiscoveredInterfaces := ptpDiscoveredInterfaceList(nodePtpDeviceAPIPath + pod.Spec.NodeName)
+					for _, inter := range ptpNotSupportedInt {
+						Expect(ptpDiscoveredInterfaces).ToNot(ContainElement(inter), fmt.Sprint("The interfaces discovered incorrectly. PTP non supported Interfaces in list"))
+					}
+				}
+			})
+
+			// 25859
+			It("PTP discovery should exclude default network interface of nodes", func() {
+				for _, pod := range ptpRunningPods {
+					var defaultRouteInterface string
+
+					Eventually(func() error {
```

Comment:
One question: why we need eventually for these commands? The pods are running, so what is the reason why the default route will appear after some time? Can't we just execute it once?

Reply:
Sometimes It couldn't exec command correctly or connect to the pod. I'm not sure why, probably libvirt issues

Reply:
Ok then, if this makes the test more robust.